docs/src/integrity/integrity.rst

```diff
@@ -0,0 +1,60 @@
+***********************
+Computational Integrity
+***********************
+
+The integrity module of Opaque ensures that the untrusted job driver hosted on the cloud service schedules tasks in the manner computed by Spark's Catalyst query optimizer. 
+Opaque runs on Spark, which utilizes data partitioning to speed up computation. 
+Specifically, Catalyst will compute a physical query plan for a given dataframe query and delegate Spark workers (run on enclaves) to compute Spark SQL operations on data partitions. 
+Each of these individual units is trusted, but the intermediary steps in which the units communicate is controlled by the job driver, running as untrusted code in the cloud. 
+The integrity module will detect if the job driver has deviated from the query plan computed by Catalyst.
```

Comment:
I think it can detect more than just whether the driver has deviated from the query plan -- also e.g. whether an adversary is shuffling data in an unexpected manner across partitions

Reply:
See new commit